rotkehlchen/assets/utils.py

```diff
@@ -94,10 +111,19 @@ def get_or_create_evm_token(
             protocol=protocol,
             underlying_tokens=underlying_tokens,
         )
-        # This can but should not raise InputError since it should not already exist
-        ethereum_token = add_ethereum_token_to_db(token_data)
-        with userdb.user_write() as cursor:
-            userdb.add_asset_identifiers(cursor, [ethereum_token.identifier])
+
+        if len(info) != 0 and asset_exists is True:
+            # This means that we might need to update the information in the database
+            GlobalDBHandler().edit_evm_token(token_data)
+        elif asset_exists is False:
+            # This can but should not raise InputError since it should not already exist
+            ethereum_token = add_ethereum_token_to_db(token_data)
+            with userdb.user_write() as cursor:
+                userdb.add_asset_identifiers(cursor, [ethereum_token.identifier])
+
+    if protocol is not None and ethereum_token.protocol != protocol:
+        object.__setattr__(ethereum_token, 'protocol', protocol)
+        GlobalDBHandler().edit_evm_token(entry=ethereum_token)
```

Comment:
Why was this needed? Is an edit what is required? what about modifying the protocol in the object but not in the database?

Reply:
Scenario that this condition addresses is:
1. In <=1.25.3 user wanted to track a curve pool token, but rotki didn't have it, so he added it manually and of course didn't specify the protocol
2. In 1.26.0 when we query curve pools, we want to set protocol to be able to properly handle price query in inquirer

Does this make sense?

Reply:
okey, then seems fine. The edit won't happen always and only when required and the protocol is not set so okey for me.

Reply:
I am still not sure what the PR does and why these changes.

The scenario @nebolax is describing here makes no sense.
You are modifying a very basic function used everywhere around the codebase and making it much more complicated.

You do this in order to handle a single edge case that sounds like data that should not be modified.

If a user breaks their DB that's their own fault, we can't have extra checks in every single DB related function trying to provide solutions. There is so many cases where user can do stuff to the DB they should not and we may break.

**Now for the scenario you describe** if you want to have it fixed the approach to have is a DB upgrade. Make sure that for all curve protocols (we know the addresses) the protocol is set properly during a DB upgrade.

Once that's done there there is no reason for this check anymore.

Reply:
@LefterisJP the problem is that we don't know the pools during the upgrade.